test/contracts/OVM/bridge/OVM_L1CrossDomainMessenger.spec.ts

```diff
@@ -89,9 +103,10 @@ describe('OVM_L1CrossDomainMessenger', () => {
 
   let OVM_L1CrossDomainMessenger: Contract
   beforeEach(async () => {
-    OVM_L1CrossDomainMessenger = await Factory__OVM_L1CrossDomainMessenger.deploy(
-      AddressManager.address
-    )
+    const xDomainMessenerImpl = await Factory__OVM_L1CrossDomainMessenger.deploy()
+    // We use an upgradable proxy for the XDomainMessenger--deploy & set up the proxy.
+    OVM_L1CrossDomainMessenger = await deployProxyXDomainMessenger(AddressManager, xDomainMessenerImpl)
+    await OVM_L1CrossDomainMessenger.initialize(AddressManager.address)
```

Comment:
I'm using this test suite to sanity check the proxy implementation -- ideally there would be unit tests for the proxy